Archipelago.MultiClient.Net.Analyzers/Archipelago.MultiClient.Net.Analyzers/Fixes/ItemFlagsInSwitchFixes.cs

```diff
@@ -0,0 +1,137 @@
+            Constants.DiagnosticPrefix + "003"
+        ];
+
+        public override FixAllProvider? GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            SyntaxNode? root = await context.Document.GetSyntaxRootAsync(context.CancellationToken)
+                .ConfigureAwait(false);
+            if (root == null)
+            {
+                return;
+            }
+
+            Diagnostic diagnostic = context.Diagnostics.First();
+            TextSpan span = diagnostic.Location.SourceSpan;
+            SwitchStatementSyntax? switchStatement = root.FindToken(span.Start).Parent?
+                .FirstAncestorOrSelf<SwitchStatementSyntax>();
+            if (switchStatement == null)
+            {
+                return;
+            }
+
+            List<SyntaxNode> caseTargets = new();
+            foreach (SwitchSectionSyntax switchSection in switchStatement.Sections)
+            {
+                IEnumerable<SyntaxNode> nodesToMove = switchSection.Labels
+                    .OfType<CaseSwitchLabelSyntax>()
+                    .SelectMany(label => label.DescendantNodes());
+
+                caseTargets.AddRange(nodesToMove);
+            }
+
+            context.RegisterCodeFix(
+                    CodeAction.Create(
+                        title: "Convert ItemFlags switch to if/else",
+                        createChangedDocument: c => ConvertHasFlagsSwitchToIfElse(
+                            document: context.Document,
+                            oldRoot: root,
+                            switchStatement: switchStatement,
+                            cancellationToken: c
+                        ),
+                        equivalenceKey: FixKeyConvertItemFlagsSwitch
+                    ),
+                    diagnostic
+                );
+        }
+
+        private async Task<Document> ConvertHasFlagsSwitchToIfElse(
+            Document document,
+            SyntaxNode oldRoot,
+            SwitchStatementSyntax switchStatement,
+            CancellationToken cancellationToken)
+        {
+            DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken);
+            var switchSections = editor.Generator.GetSwitchSections(switchStatement);
+            var switchExpression = switchStatement.Expression;
+
+            // This should be an empty SyntaxList based on the default.
+            SyntaxList<StatementSyntax> @else = default;
+            SyntaxNode current = null;
+
+            // Iterating in reverse order to generate the if-statement chain from the bottom up.
+            foreach (var node in switchSections.Reverse())
+            {
+                SwitchSectionSyntax switchSection = (SwitchSectionSyntax)node;
+                List<ExpressionSyntax> cases = switchSection.Labels.OfType<CaseSwitchLabelSyntax>().Select(x => x.Value).ToList();
```

Comment:
I think we might have to watch out for pattern matching here, for most kinds of patterns it would be invalid syntax anyway but e.g. the user could potentially get away with something like `case var f when f.HasFlag(ItemFlags.Advancement)` which is not a violation of the rule but would result in this fix producing bad syntax

Reply:
Cases with bitwise | could also be a concern (assuming they are legal) in that maybe in that case the expected comparison actually should be ==